src/internal/packager2/load.go

```diff
@@ -107,82 +93,103 @@ func identifySource(src string) (string, error) {
 	if strings.Contains(src, ".part000") {
 		return "split", nil
 	}
+	// match deployed package names: lowercase, digits, hyphens
+	namePattern := regexp.MustCompile(`^[a-z0-9][a-z0-9\-]*$`)
```

Comment:
Would it cause a cyclical dependency to use `lint.IsLowercaseNumberHyphenNoStartHyphen` here? It's also generally preferred to run `regexp.MustCompile` as top level variables that way on the off chance they fail it'll be found immediately.